File: solutions/ocaml/difference-of-squares/1/difference_of_squares.ml
let square_of_sum n =
  let rec sum acc n = if n = 0 then acc else sum (acc + n) (n - 1) in
  let sum = sum 0 n in
  sum * sum

let sum_of_squares n =
  let rec sum acc n = if n = 0 then acc else sum (acc + (n * n)) (n - 1) in
  sum 0 n

let difference_of_squares n = square_of_sum n - sum_of_squares n

File: solutions/cpp/difference-of-squares/1/difference_of_squares.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {
    int square_of_sum(int num) {
        return pow((num * (num + 1)) / 2, 2);
    }

    int sum_of_squares(int num) {
        return (num * (num + 1) * ((num * 2) + 1)) / 6;
    }

    int difference(int num) {
        int sum, square_sum;

        sum = (num * (num + 1)) / 2;
        square_sum = (num * (num + 1) * ((num * 2) + 1)) / 6;

        return (sum * sum) - square_sum;
    }
}  // namespace difference_of_squares
